CORE/world_gadgets/data_engine.rb

```diff
@@ -0,0 +1,47 @@
+
+
+class DataEngine
+
+  def initialize(logger=nil)
+    @logger = logger
+  end
+
+  def load_data_from_yml(filename)
+    @logger.debug "Loading data from file #{filename}" if @logger
+    begin
+      data = YAML.load_file(filename)
+      return data
+    rescue Psych::SyntaxError => e
+      raise "ERROR: Tally could not parse this YML file!\n"\
```

Comment:
I will change the Tally references. As for the method length, it's really not that long, the string constants just make it seem so I think.